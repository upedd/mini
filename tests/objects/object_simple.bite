native print;

object Foo {
    bar() {
        print(foo());
    }

    foo() {
        return x;
    }

    x = 10;

    _y;
    get y {
        print("Foo.y");
        return _y;
    }
    set y(value) {
        print("=Foo.y");
        _y = value;
    }
}

print(Foo.x);
print(Foo.foo());
Foo.bar();
Foo.x = 20;
print(Foo.x);
print(Foo.foo());
Foo.bar();

Foo.y = 10;
print(Foo.y);
Foo.y = 20;
print(Foo.y);