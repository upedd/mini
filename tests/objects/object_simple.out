10
10
10
20
20
20
=Foo.y
Foo.y
10
=Foo.y
Foo.y
20